File: code/c/procedure-decl/print-steps.c
/*
* Program: print_steps.c
* Prints out the steps to cook a meal... (partial)
*/

void find_what_to_cook()
{
    printf("Step 1 - Find what to cook -\n");
    printf("1 - Find a recipe book\n");
    printf("2 - Pick recipe \n");
}

void purchase_missing_ingredients()
{
  printf("Step 2(a) - Purchase Missing Ingredients -\n");
  printf("1 - Goto Shop\n");
  printf("2 - Find ingredients and put in basket\n");
  printf("3 - Go to register and pay for ingredients in basket\n");
  printf("4 - Return home\n");
}

void get_ingredients()
{
  printf("Step 2 - Purchase Ingredients -\n");
  printf("1 - Read recipe\n");
  printf("2 - Write a list of ingredients \n");
  printf("3 - Check food stocks, and tick off ingredients you already have\n");
  purchase_missing_ingredients();
}

int main()
{
    find_what_to_cook();
    get_ingredients();
    // etc...
    
    return 0;
}

File: code/pascal/procedure-decl/PrintSteps.pas
program PrintSteps; // Prints out the steps to cook a meal... (partial)

procedure FindWhatToCook();
begin
    WriteLn("Step 1 - Find what to cook -");
    WriteLn("1 - Find a recipe book");
    WriteLn("2 - Pick recipe ");
end;

procedure PurchaseMissingIngredients();
begin
  WriteLn("Step 2(a) - Purchase Missing Ingredients -");
  WriteLn("1 - Goto Shop");
  WriteLn("2 - Find ingredients and put in basket");
  WriteLn("3 - Go to register and pay for ingredients in basket");
  WriteLn("4 - Return home");
end;

procedure GetIngredients();
begin
  WriteLn("Step 2 - Purchase Ingredients -");
  WriteLn("1 - Read recipe");
  WriteLn("2 - Write a list of ingredients ");
  WriteLn("3 - Check food stocks, and tick off ingredients you already have");
  PurchaseMissingIngredients();
end;

procedure Main();
begin
    FindWhatToCook();
    GetIngredients();
    // etc...
end;

begin
    Main();
end.